cache/cache.c

```diff
@@ -0,0 +1,101 @@
+#include "cache.h"
+#include "./config.h"
+
+/* private data used by cache_t */
+typedef struct {
+  redisContext* rc;
+} connection_private;
+#define CONN(c) ((connection_private*)(c->conn))
+
+/*
+ * Private functions
+ */
+
+int redis_del(redisContext* c, const char* const key) {
+  int ret = 1;
+
+  if (key == NULL) {
+    return ret;
+  }
+
+  redisReply* reply = redisCommand(c, "DEL %s", key);
+  if (reply->integer) {
+    ret = 0;
+  }
+
+  freeReplyObject(reply);
+  return ret;
+}
+
+int redis_get(redisContext* c, const char* const key, char* res) {
+  int ret = 1;
+
+  if (key == NULL) {
```

Comment:
Check `res` here

Reply:
oops, just followed ditto but forgot this.